core/base/ftmTree/FTMTreeUtils_Template.h

```diff
@@ -10,44 +10,35 @@
 
 #pragma once
 
-//#include <FTMTreeMT.h>
-
 namespace ttk {
   namespace ftm {
     // --------------------
-    // Get
+    // Utils
     // --------------------
+    template <typename type>
+    static type myAbs(const type var) {
+      return (var >= 0) ? var : -var;
+    }
 
     template <class dataType>
-    std::tuple<dataType, dataType> FTMTree_MT::getBirthDeath(idNode nodeId) {
-      idNode originId = this->getNode(nodeId)->getOrigin();
-      if(this->isNodeOriginDefined(
-           nodeId)) { // Avoid error if origin is not defined
-        dataType pers1 = this->getValue<dataType>(nodeId);
-        dataType pers2 = this->getValue<dataType>(originId);
-        dataType birth = std::min(pers1, pers2);
-        dataType death = std::max(pers1, pers2);
-        return std::make_tuple(birth, death);
-      }
-      return std::make_tuple(0.0, 0.0);
+    bool isEqual(dataType first, dataType two, double eps = 1e-6) {
+      return myAbs<dataType>(first - two)
+             < eps * std::max(myAbs<dataType>(first), myAbs<dataType>(two));
     }
 
     template <class dataType>
-    dataType FTMTree_MT::getBirth(idNode nodeId) {
-      return std::get<0>(this->getBirthDeath<dataType>(nodeId));
+    bool isInferior(dataType first, dataType two) {
```

Comment:
why isn't the operator `<` sufficient here?

Reply:
I think it is the same reason than for the `isEqual` function. Two double can be considered as not equal because a far decimal can be different and causing the inequality condition to be true while the two numbers are globally the same.

Reply:
to replace by `>` or `<` and check against all the state files of the paper

Reply:
Done in the next commit!